_Group Projects/Group_2/Week_8/program.do

```diff
@@ -0,0 +1,74 @@
+cap program drop pmt 
+clear all
+
+program define pmt, rclass // define an r-class program so that the program writes into return and we can get stuff out of the program
+
+// generate the cutoff for the bottom quintile of 30 observations
+set obs 18 // our intervention has 18 strata 
+gen cluster_id = _n 
+expand 30 // creating 30 duplicates for each cluster 
+bysort cluster_id: gen household_id = _n // creating a household id for each cluster
+
+/* we are generating separate levels of household income based on their strata. 
+For the ease of this simulation, there are broadly three kinds of households. 
+Poor (the first six clusters), middle income (cluster 7-12), and rich (clusters 13-18)
+A similar approach is also adopted for the education and family size variables*/
+
+gen inc_origin = . 
+replace inc_origin = exp(rnormal()) if inrange(cluster_id, 1, 6)
+replace inc_origin = exp(rnormal(1, 0.1)) if inrange(cluster_id, 7, 12)  
+replace inc_origin = exp(rnormal(2, 0.1)) if inrange(cluster_id, 13, 18)
+replace inc_origin = abs(inc_origin)
+
+gen educ_yrs = . 
+replace educ_yrs = floor(rnormal(5,1)) if inrange(cluster_id, 1, 4)
+replace educ_yrs = floor(rnormal(7.4,1)) if inrange(cluster_id, 5, 6)
+replace educ_yrs = floor(rnormal(10,1)) if inrange(cluster_id, 7, 12)
+replace educ_yrs = floor(rnormal(14,1)) if inrange(cluster_id, 13, 18)
+replace educ_yrs = abs(educ_yrs)
+
+gen age = trunc(rnormal(40,2))
+drop if age < 15
+gen age_sq = age*age 
+
+gen work_exp = trunc(rnormal(10,2))
```

Comment:
I'm not sure it's entirely necessary, but would have been cool to generate the `work_exp` variable such that it's correlated with the `educ_years` variable, given that probably mimics those two variables relationship in real life! (I guess same could be said of `age` and `fam_size` variables, `age` and `educ_year` variables, etc.). Doesn't seem totally necessary but wondered if that might be helpful for our next assignment!

Reply:
Hmm, yes! I see why we should be doing that. Will play around with this when I am revising the do-files!